doc: Describe CAF module suspend/resume request events support (#23556)
nrfconnect/sdk-nrf: applications/nrf_desktop/doc/usb_state_pm.rst

* If the USB state is set to :c:enum:`USB_STATE_POWERED`, the module restricts the power down level to the :c:enum:`POWER_MANAGER_LEVEL_SUSPENDED`.
* If the USB state is set to :c:enum:`USB_STATE_ACTIVE`, the :c:enum:`POWER_MANAGER_LEVEL_ALIVE` is required.
* If the USB state is set to :c:enum:`USB_STATE_SUSPENDED`, the :c:enum:`POWER_MANAGER_LEVEL_SUSPENDED` is imposed.
  The module restricts the power down level to the :c:enum:`POWER_MANAGER_LEVEL_SUSPENDED`.
  The module also submits a :c:struct:`force_power_down_event` to force a quick power down.
* If the USB state is set to :c:enum:`USB_STATE_DISCONNECTED`, any power level is allowed.
  While disconnecting the USB cable, the :c:enum:`USB_STATE_SUSPENDED` USB state might be reported before the :c:enum:`USB_STATE_DISCONNECTED` USB state.

>   For application to behave consistently regardless of whether the :c:enum:`USB_STATE_SUSPENDED` USB state was reported, the module also submits a :c:struct:`force_power_down_event` to force a quick power down.

Reviewer: ```suggestion
  For the application to behave consistently regardless of whether the :c:enum:`USB_STATE_SUSPENDED` USB state was reported, the module also submits a :c:struct:`force_power_down_event` to force a quick power down.
```

>   For the application to behave consistently regardless of whether the :c:enum:`USB_STATE_SUSPENDED` USB state was reported, the module also submits a :c:struct:`force_power_down_event` to force a quick power down.

Reviewer 2: consider rephrasing

Author: I know that it's a bit complicated, but I don't see any way on how to improve it.

  The module initially restricts the power down level to the :c:enum:`POWER_MANAGER_LEVEL_SUSPENDED`.
  Then, after the :ref:`CONFIG_DESKTOP_USB_PM_RESTRICT_REMOVE_DELAY_MS <config_desktop_app_options>` configurable delay, the module removes the power down level restriction.

>   This is done to allow user to take actions, like e.g. restart Bluetooth LE advertising, after disconnecting the USB cable without going through reboot.

Reviewer: ```suggestion
  This is done to allow you to take actions, such as restart Bluetooth LE advertising, after disconnecting the USB cable without going through reboot.
```

>   This is done to allow you to take actions, such as restart Bluetooth LE advertising, after disconnecting the USB cable without going through reboot.

Reviewer: ```suggestion
  This allows you to take actions, such as restart Bluetooth LE advertising, after disconnecting the USB cable without going through reboot.
```
